server/src/main/java/org/elasticsearch/search/fetch/FetchContext.java

```diff
@@ -207,4 +206,48 @@ public ScriptFieldsContext scriptFields() {
     public SearchExtBuilder getSearchExt(String name) {
         return searchContext.getSearchExt(name);
     }
+
+    /**
```

Comment:
I put all of there here rather than calling `getQueryShardContext().fieldType` because there is some confusion around what is "ok" to call where. Putting these are methods does two things:
1. It makes us talk about whether or not it is "ok" to use QueryShardContext here.
2. If we decide that it is, then we have a method folks can use that they know is "ok".